pkg/util/util.go

```diff
@@ -75,6 +75,38 @@ type OAuth2Endpoints struct {
 	TokenEndpoint         string `json:"token_endpoint"`
 }
 
+// ValidationError is a custom error if the validation failed
+type ValidationError struct {
+	Msg string
+}
+
+// TechnicalError is a custom error if an operation in the validation process failed
+type TechnicalError struct {
```

Comment:
I don't think there is a need for a `TechnicalError` type. it will just force you to catch all non-validation errors and return them as a TecnicalError